### ISSUE TITLE
Crash to desktop when launching MegaDrive roms using BlastEm Core on Mac OS with Intel CPU and GPU

### ISSUE THREAD

**Issue by Sethur** [libretro/RetroArch#15703]
## Description

When launching any Sega MegaDrive ROM using BlastEm and default settings on a fresh install, I get a crash to desktop without any error message. I tried all available video drivers and got the same error every time.

BlastEm was working fine on a previous version with my hardware.

### Steps to reproduce the bug

1. Do a fresh install of RetroArch on an Intel Mac with Intel GPU (OS version Monterey 12.6.7)
2. Launch any Sega MegaDrive ROM with BlastEM core
3. Experience immediate crash to desktop

### Bisect Results

This issue started sometime during the last year. It was working fine with previous versions of RetroArch. I don't think this is a core bug, as BlastEm was apparently not updated for a long time.

### Version/Commit
RetroArch: 1.16.0  Build Aug 20, 2023

### Environment information

- OS: Mac OS Monterey 12.6.7

**Comment by sonninnos** [comment]
Core issues belong in the core repo, not here.

**Comment by Sethur** [comment]
From the cores I use, I only get a CTD with BlastEm, other cores work fine. And as I said, BlastEm was working fine with a previous version of RetroArch. It is probably not a core issue, since the core wasn't updated for a long time. I also posted this issue in the BlastEm core, but I don't think it is actively developed any more.

When I launch Retroarch via the CLI, when the CTD happens, the following message is shown:

```
alloc_code: Cannot allocate memory
Failed to allocate memory for generated code
```

**Comment by Sethur** [comment]
Can anybody reproduce this on an Intel Mac?

**Comment by fmaida** [comment]
Yes. I'm having the very same problem on my MacBook Air with an old Intel i5 processor and MacOS 12.7.3

```
Product ID: 76334511
SHA1: ca2c05c070dac4b2ad7f2425ccad1b7d65949e49
Not found in ROM DB, examining header

alloc_code: Cannot allocate memory
Failed to allocate memory for generated code
```
